docs/esp32/tutorial/intro.rst

```diff
@@ -36,106 +36,95 @@ Getting the firmware
-device before putting on new MicroPython firmware.
+* Download a stable firmware release.
+* Download a daily firmware "Preview" build.
 
-Currently we only support esptool.py to copy across the firmware.  You can find
-this tool here: `<https://github.com/espressif/esptool/>`__, or install it
-using pip::
+If you are just starting with MicroPython, the best bet is to go for the stable
+Release firmware builds. If you are an advanced, experienced MicroPython ESP32
+user who would like to follow development closely and help with testing new
+features, then you may find the Preview builds useful.
 
-    pip install esptool
-
-Versions starting with 1.3 support both Python 2.7 and Python 3.4 (or newer).
-An older version (at least 1.2.1 is needed) works fine but will require Python
-2.7.
+.. _esp32_flashing:
 
-Using esptool.py you can erase the flash with the command::
+Deploying the firmware
+----------------------
 
-    esptool.py --port /dev/ttyUSB0 erase_flash
+Once you have the MicroPython firmware you need to load it onto your ESP32
+device. There are two main steps to do this: first you need to put your device
+in bootloader mode, and second you need to copy across the firmware. The exact
+procedure for these steps is highly dependent on the particular board.
 
-And then deploy the new firmware using::
+Detailed steps can be found on the same `MicroPython download page`_ for your
+board. It's recommended that you follow the steps on the download page, as they
+are customised for your particular board.
 
-    esptool.py --chip esp32 --port /dev/ttyUSB0 write_flash -z 0x1000 esp32-20180511-v1.9.4.bin
+If the above commands run without error then MicroPython should be installed on
+your board! Skip ahead to :ref:`esp32_serial_prompt`.
 
-Notes:
+.. _esp32_troubleshooting_install:
 
-* You might need to change the "port" setting to something else relevant for your
-  PC
-* You may need to reduce the baudrate if you get errors when flashing
-  (eg down to 115200 by adding ``--baud 115200`` into the command)
-* For some boards with a particular FlashROM configuration you may need to
-  change the flash mode (eg by adding ``-fm dio`` into the command)
-* The filename of the firmware should match the file that you have
+Troubleshooting installation problems
+-------------------------------------
 
-If the above commands run without error then MicroPython should be installed on
-your board!
+If you experience problems during flashing or with running firmware immediately
+after flashing, here are some troubleshooting recommendations:
+
+* Esptool will try to detect the serial port where your ESP32 is connected. If
+  this doesn't work, or you have multiple serial ports, then you may need to
+  manually specify the port by adding the ``--port`` option to the start of the
+  ``esptool.py`` command line. For example, ``esptool.py --port /dev/ttyUSB0
+  <rest of line>`` for Linux or ``esptool.py --port COM4 <rest of line>`` for
```

Comment:
```suggestion
  <rest of line>`` for Linux or ``esptool --port COM4 <rest of line>`` for
```

Reply:
Thanks @mattytrentini. Good catch on the command name here, have applied.